### PR TITLE
Use Azure REST API instead of `az` CLI

### PR DESCRIPTION
**Pull request by bear454**
Due to maintenance issues with the Azure CLI & SLE, using the REST API is preferable. This is a complete behind-the-scenes transition with no UI or workflow changes.

### CODE REVIEW

File: engines/azure/app/models/azure/interface.rb [hunk 1]
```diff
@@ -0,0 +1,358 @@
+module Azure
+  require 'base64'
+
+  class Interface
+    class LoginError < StandardError; end
+
+    RESOURCE = 'https://management.azure.com/'
+    HOST = 'https://management.azure.com'
```
Comment (jesusbv) [inline engines/azure/app/models/azure/interface.rb:8]: `RESOURCE` and `HOSTS` are the same, there should be only one

File: engines/azure/app/models/azure/interface.rb [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    # REST wrappers
+
+    def api_get(path:, params: {}, capture_exception: false)
+      self.login() if @token.blank?
+      response = api_exception_trap(capture_exception) do
+        RestClient.get(
+          HOST + "/subscriptions/#{@subscription.value}" + path,
+          {
+            params: {
+              'api-version' => '2023-07-01',
```
Comment (jesusbv) [inline engines/azure/app/models/azure/interface.rb:77]: could we use a const for the repeating api version values ?

File: engines/azure/app/models/azure/interface.rb [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+    end
+
+    def api_head(path:, params: {}, capture_exception: true)
+      self.login() if @token.blank?
+      api_exception_trap(capture_exception) do
+        RestClient.head(
+          HOST + "/subscriptions/#{@subscription.value}" + path,
+          {
+            params: {
+              'api-version' => '2023-07-01',
```
Comment (jesusbv) [inline engines/azure/app/models/azure/interface.rb:94]: are these versions the only version where the endpoint works ?
or the minimum one ?
could other endpoints use this version as well ?

File: engines/azure/app/models/azure/interface.rb [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+          'api-version' => '2018-05-01'
+        }
+      )
+    end
+
+    def destroy_dns_record(resource_group:, record_type:, record:, domain:)
+      api_delete(
+        path: "/resourceGroups/#{ resource_group }/providers/Microsoft.Network/dnsZones/#{ domain }/#{ record_type }/#{ record }",
+        params: {
+          'api-version' => '2018-05-01'
```
[the remark this answers is not in the document]
Comment (jesusbv) [reply engines/azure/app/models/azure/interface.rb:305]: there is an azure cli command to retrieve/fetch the api versions available
```bash
> az provider show --namespace Microsoft.Network --query "resourceTypes[?resourceType=='networkProfiles'].apiVersions | [0]" --out table
Result
----------
2023-05-01
2023-04-01
...
2018-05-01
>
```

[the remark this answers is not in the document]
Comment (jesusbv) [reply engines/azure/app/models/azure/interface.rb:305]: meaning there should be an endpoint under the hood to fetch that info

File: engines/azure/app/models/azure/interface.rb (hunk 1 above)
Comment (bear454) [reply engines/azure/app/models/azure/interface.rb]: Fixed in 9f23e7d . Thanks for catching that.

File: engines/azure/app/models/azure/interface.rb (hunk 2 above)
Comment (bear454) [reply engines/azure/app/models/azure/interface.rb]: How about 63ade51 ?

File: engines/azure/app/models/azure/interface.rb (hunk 3 above)
Comment (bear454) [reply engines/azure/app/models/azure/interface.rb]: A version must be supplied; and the versions listed are clearly documented in Microsoft documentation, and function as documented.

File: engines/azure/app/models/azure/interface.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (bear454) [reply engines/azure/app/models/azure/interface.rb]: We cannot dynamically select the API version; there are undocumented changes in newer versions.

**Pull request merged**